model/src/main/java/com/epam/esm/task/dao/impl/GiftCertificateDaoImpl.java

```
@@ -0,0 +1,178 @@
+        }
+    }
+
+    @Override
+    public List<GiftCertificate> findByColumn(String columnName, String data) {
+        String query = creator.findByColumn(tableName,columnName);
+        return jdbcTemplate.query(query,new GiftCertificateMapper(builder),data);
+    }
+
+    public List<GiftCertificate> findByParam(Map<String, String> validParam) {
+        List<GiftCertificate> result = jdbcTemplate.query(
+                creator.findByParam(tableName,validParam),
+                new GiftCertificateMapper(builder));
+        findByListGift(result);
+        return result;
+    }
+
+    @Override
+    protected long executeEntity(GiftCertificate entity, String query) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(connection -> {
+            PreparedStatement statement = connection.
+                    prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            fillPreparedStatement(entity,statement);
+            return statement;
+        },keyHolder);
+        return keyHolder.getKey().longValue();
+    }
+
+    @Override
+    protected void fillPreparedStatement(GiftCertificate entity,PreparedStatement statement) throws SQLException {
+        statement.setString(1, entity.getName());
+        statement.setString(2,entity.getDescription());
+        statement.setBigDecimal(3,entity.getPrice());
+        statement.setInt(4, entity.getDuration());
+        statement.setDate(5,Date.valueOf(entity.getCreate_date().toLocalDate()));
+        statement.setDate(6,Date.valueOf(entity.getUpdate_date().toLocalDate()));
+    }
+
+    private void findByListGift(List<GiftCertificate> gifts){
+        gifts.forEach(i -> {
+            List<Tag> tagList = new ArrayList<>();
+            mtmDao.findByGiftId(i.getId()).forEach( j -> {
+
+                try {
+                    tagList.add(tagDao.findById(j.getTagId()));
+                } catch (DaoException ignored) {}
```

Ups.... Why are you suppressing exception?